docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -0,0 +1,454 @@
+
+This order means that a custom converter in the `Converters` collection is overridden by a converter that is registered by applying an attribute at the type level. Both of those registrations are overridden by an attribute at the property level.
+
+The `System.Text.Json` registration precedence for custom converters is different:
+
+* Attribute on property
+* <xref:System.Text.Json.JsonSerializerOptions.Converters> collection
+* Attribute on type
+
+The difference here is that a custom converter in the `Converters` collection overrides an attribute at the type level. The intention behind this order of precedence is to make run-time changes override design-time choices. There's no way to change the precedence.
+
+### Character escaping
+
+During serialization, `Newtonsoft.Json` is relatively permissive about letting characters through without escaping them. (That is, it doesn't replace them with `\uxxxx` where `xxxx` is the character's code point). Where it does escape them, it does so by emitting a `\` before the character (for example, `"` becomes `\"`). `System.Text.Json` escapes more characters by default to provide defense-in-depth protections against cross-site scripting (XSS) or information-disclosure attacks and does so by using the six-character sequence. `System.Text.Json` escapes all non-ASCII characters by default, so you don't need to do anything if you're using `StringEscapeHandling.EscapeNonAscii` in `Newtonsoft.Json`. `System.Text.Json` also escapes HTML-sensitive characters, by default. For information about how to override the default `System.Text.Json` behavior, see [Customize character encoding](system-text-json-how-to.md#customize-character-encoding).
+
+### Deserialization of Object properties
+
+When `Newtonsoft.Json` deserializes to `Object` properties in POCOs or in dictionaries of type `Dictionary<string, Object>`, it:
+
+* "Guesses" the type of "primitive" values in the JSON and stores `string`/`long`/`double`/`boolean`/`DateTime` as a boxed object.
+* Stores "complex" values as `JObject`.
+* Stores `null` when the payload has "null".
+
+`System.Text.Json` stores a boxed `JsonElement` for "primitive" and "complex" values within the System.Object property or dictionary value. It stores `null` when the payload has "null" in it.
+
+To implement type inference for `Object` properties, create a converter like the example in [How to write custom converters](system-text-json-converters-how-to.md#deserialize-inferred-types-to-object-properties).
+
+### Maximum depth
+
+`Newtonsoft.Json` doesn't have a maximum depth limit. For `System.Text.Json` there's a default limit  of 64, and it's configurable by setting (xref:System.Text.Json.JsonSerializerOptions.MaxDepth?displayProperty=nameWithType).
+
+## Scenarios using JsonSerializer that require workarounds
+
+The following scenarios aren't supported by built-in functionality, but sample code is provided for workarounds.
```

Comment:
We should mention that most workarounds require writing custom converters up front here since that seems to be the common theme.